src/Chapter3.hs

```diff
@@ -970,6 +1092,28 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
+data DaysOfWeek = Monday 
+                  | Thuesday 
+                  | Wednesday 
+                  | Thursday 
+                  | Friday 
+                  | Saturday 
+                  | Sunday deriving (Show, Eq, Enum)
+
+isWeekend :: DaysOfWeek -> Bool
+isWeekend Saturday = True
+isWeekend Sunday = True
+isWeekend _ = False
+
+nextDay :: DaysOfWeek -> DaysOfWeek
+nextDay d
+    | d == Sunday = Monday
+    | otherwise = succ d
+
+daysToParty :: DaysOfWeek -> Int
+daysToParty d = length datyToFriday - 1
+                  where
+                    datyToFriday = enumFromTo d Friday
```

Comment:
That's a correct implementation 👍🏻 
If you want to make it more efficient, you can use the `fromEnum` method of the `Enum` typeclass